terraform/modules/application/alarms.tf

```diff
@@ -115,3 +115,41 @@ resource "aws_cloudwatch_metric_alarm" "sidekiq_memory" {
     ServiceName = aws_ecs_service.sidekiq.name
   }
 }
+
+resource "aws_cloudwatch_metric_alarm" "sidekiq_tasks_exited" {
+  alarm_actions             = [var.sns_topic_arn]
+  alarm_name                = "${var.prefix}-sidekiq-tasks-exited"
+  comparison_operator       = "GreaterThanOrEqualToThreshold"
+  datapoints_to_alarm       = 1
+  evaluation_periods        = 1
+  metric_name               = "TriggeredRules"
+  namespace                 = "AWS/Events"
+  ok_actions                = [var.sns_topic_arn]
+  period                    = 3 * 60 # By causing sidekiq to fail repeatedly (with both 1 tasks and 2 tasks running) I found that double the desired count would reliably fail in a 3 minute period.
```

Comment:
In order to test this I updated the command in the container definition to be this: `command = ["echo", "'Testing failure!'", "&&", "exit", "1"]`. (I then triggered a deployment from the core repo (branch `ForceAwsTaskFailure`) to ensure the service was pulling a valid image.)

Reply:
Here are the container starts with desired_count = 1

(I stopped this after about 10 minutes)
![image](https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data-infrastructure/assets/63662292/59a68a00-04b8-44ac-9fc5-7a2bbd617e21)

Reply:
Here are the container starts with desired_count = 2. I also used this to decide on the period.

(I also stopped this after about 10 minutes)
![image](https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data-infrastructure/assets/63662292/e4abbc29-3346-4758-86f7-493cfb69ee78)